lib/rjgit.rb

```diff
@@ -190,6 +190,40 @@ def self.describe(repository, ref, options = {})
       end
       command.call
     end
+
+    # options:
+    #  * ref
+    #  * path_filter
+    #  * case_insensitive
+    def self.grep(repository, query, options={})
+      case_insensitive = options[:case_insensitive]
+      repo = RJGit.repository_type(repository)
+      walk = RevWalk.new(repo)
+      ls_tree_options = {:recursive => true, :path_filter => options[:path_filter]}
+
+      raise "A #{query.class} was passed to #{self}.grep().  Only Regexps and Strings are supported!" unless String === query || Regexp === query
+
+      if case_insensitive
+        query = case query
+        when Regexp then Regexp.new(query.source, query.options | Regexp::IGNORECASE)
+        when String then query.downcase
+        end
```

Comment:
I meant add an `else` clause here. Raising an exception should still work even though the `case` statement is used in an assignment:

```ruby
2.6.3 :001 > query = raise "FAIL"
Traceback (most recent call last):
[...]
RuntimeError (FAIL)
```